### DISCUSSION TITLE
enqueueShortcut with delay stopps working after some seconds

### DISCUSSION THREAD

**Discussion by radi5** [Waboodoo/HTTP-Shortcuts#374] [Q&A]
Hi team, 
I need some help. I automated my garage using shortcuts and it works really fine. For updating the status I use
`enqueueShortcut(<uid>,null,25*1000`. 
This works fine if started out of the shortcut app itself. If I use a quick button from homescreen for it, this enqueued shortcut is only executed if timer is set to 10 or less. If I open shortcut app it is immediately executed. I guess it is a kind of sleep mechanism that stops shortcuts from executing something when not running activly anymore.

The workaround
`wait(25*1000)
executeShortcut(<uid>)`
works, but is not as good as enqueue as it blocks the app.
Has anyone any idea?
Thanks in advance.

radi

**Comment by Waboodoo** [comment]
Hi radi,

As a first step of troubleshooting, you could try excluding the app from the device's battery optimization. You can find this option in the Settings under the Troubleshooting section. This might solve your problem. If not, let me know, I can try to investigate further.

**Reply by radi5** [reply]
Hi @Waboodoo,
that was the first thing I did and this did not change the bahvior. Anyway I guess it has something to do with it as shortly before it stops reacting also the screen gets darker.
But what else can I do beside taking the app out of the battery optimization. Any idea how to get more details. Are there any logs that can be used for investigation?
BR radi

**Reply by Waboodoo** [reply]
I think there's not much you can do with the app as it currently is. The Android OS is trying to manage its resources and therefore sooner or later ends up stopping the app when it's not in the foreground. This is something that the app currently doesn't handle well, I'll need to look into improving that at some point. The whole concept of scheduling / delayed execution could use some more love.
All I can suggest as a workaround for now would be to delegate all the scheduling to a third-party automation app such as Tasker, MacroDroid or similar.
